src/luxor_overrides.jl

```diff
@@ -280,3 +281,254 @@ function background(background_color)
+        push!(retpoly, Point(res[1], res[2]))
+    end
+    retpoly
+end
+
+"""
+updates CURRENT_JPATHS
+if used inside strokepath() pass action=:stroke
+and inside fillpath pass :fill.
+move this to luxor_overrides_util.jl later.
+"""
+function update_currentjpath(action::Symbol)
+    if CURRENT_FETCHPATH_STATE
+        #println("test strokepaths")
+        cur_polys, cur_costates = pathtopoly(Val(:costate))
+        @assert length(cur_polys) == length(cur_costates)
+        transform = getmatrix()
+        cur_polys = [apply_transform(getmatrix(), poly) for poly in cur_polys]
+        #cur_polys is of 2 element Tuple 
+        #containg 2 arrays 1 with Polygons and one with the bools
+        r, g, b, a = map(
+            sym -> getfield(Luxor.CURRENTDRAWING[1], sym),
+            [:redvalue, :greenvalue, :bluevalue, :alpha],
+        )
+        #by default they are black transparent 
+        fillstroke = Dict(:fill => [0.0, 0, 0, 0], :stroke => [0.0, 0, 0, 0])
+        fillstroke[action] .= [r, g, b, a]
+        #if polys didnt change we just modify the last JPath in the CURRENT_JPATHS
+        if length(CURRENT_JPATHS) > 0 && cur_polys == CURRENT_JPATHS[end].polys
+            #print("found similar path\n")
+            setproperty!(CURRENT_JPATHS[end], action, fillstroke[action])
+            #if action was :stroke above line would be the same as doing...
+            #CURRENT_JPATHS[end].stroke = fillstroke[:stroke] 
+            setproperty!(CURRENT_JPATHS[end], :lastaction, action)
+            #if action was :stroke above line would be the same as doing...
+            #CURRENT_JPATHS[end].lastaction = :stroke
+        else
+            #println("adding to CURRENT_JPATH")
+            currpath = JPath(
+                cur_polys,
+                cur_costates,
+                fillstroke[:fill],
+                fillstroke[:stroke],
+                action,
+                2,
+            )
+            push!(CURRENT_JPATHS, currpath)
+        end
+    end
+end
+
+function Luxor.strokepath()
+    update_currentjpath(:stroke)
+    if !DISABLE_LUXOR_DRAW
+        Luxor.get_current_strokescale() ?
+        Luxor.Cairo.stroke_transformed(Luxor.get_current_cr()) :
+        Luxor.Cairo.stroke(Luxor.get_current_cr())
+    else
+        newpath()
+    end
+end
+
+function Luxor.strokepreserve()
+    update_currentjpath(:stroke)
+    if !DISABLE_LUXOR_DRAW
+        Luxor.get_current_strokescale() ?
+        Luxor.Cairo.stroke_preserve_transformed(Luxor.get_current_cr()) :
+        Luxor.Cairo.stroke_preserve(Luxor.get_current_cr())
+    end
+
+end
+
+function Luxor.fillpath()
+    update_currentjpath(:fill)
+    if !DISABLE_LUXOR_DRAW
+        Luxor.Cairo.fill(Luxor.get_current_cr())
+    else
+        newpath()
+    end
+end
+
+function Luxor.fillpreserve()
+    update_currentjpath(:fill)
+    if !DISABLE_LUXOR_DRAW
+        Luxor.Cairo.fill_preserve(Luxor.get_current_cr())
+    end
+end
+
+function Luxor.pathtopoly(::Val{:costate})
```

Comment:
can you add a docstring for this function? Probably also for the ones above but this one isn't that obvious to me just from the function name and argument.